### PR TITLE
docs: refactor markdown rendering and add switching …

### PR DESCRIPTION
**Pull request by HerrTopi**
…feature to examples

### CODE REVIEW

File: docs/guides/writing-docs.md [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+<Button>InstUI button</Button>
+```
+
+**_Note:_** you can use any instUI components in the examples
+
+### 3. example
+
+The most complex type is the `type: example`. It will render as the `embed` did, but it also provides access to the code, which is editable and the changes reflect on the rendered view immediately.
+
+There are to ways these examples can be written.
```
Comment (balzss) [inline docs/guides/writing-docs.md:83]: typo, should be `two`

File: package.json [hunk 2]
```diff
@@ -117,5 +117,9 @@
   },
   "browserslist": [
     "extends @instructure/browserslist-config-instui"
-  ]
+  ],
+  "dependencies": {
+    "marked-react": "^2.0.0",
+    "uuid": "^9.0.1"
+  }
```
Comment (matyasf) [inline package.json:124]: is there any reason why these need to be in the root package.json?

Comment (HerrTopi) [reply package.json]: Nope :D

File: docs/contributor-docs/api-guidelines.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 title: API Guidelines
-category: Getting Started
-order: 5
+category: Contributor Guides
+order: 4
```
Comment (matyasf) [inline docs/contributor-docs/api-guidelines.md:4]: I really like the new sidebar ordering, its much more readable for endusers IMO

File: docs/contributor-docs/writing-docs.md [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+
+# How To Write Docs
+
+This page documents how to add and maintain documentation.
+
+## Overview
+
+The documentation site is generated from the **source code** and the `.md` files. There are two types of `.md` files: `README.md` and `named-md-files.md`, under the `docs folder`.
+
+The **source code** is parsed with [JSDoc](https://jsdoc.app/) and it inprocides `type` information for the docs.
```
Comment (matyasf) [inline docs/contributor-docs/writing-docs.md:15]: its also parsed by `react-docgen` for React components.

**Review by matyasf — CHANGES_REQUESTED** [review]
When I try to open a functional example in CodeSandbox I get and error: `{"error":"Unable to process params for /define - please double check them. See https://codesandbox.io/docs/importing#create-wizard for more details."}` 
Also see my other comments.

File: packages/__docs__/src/Playground/index.tsx [hunk 5]
```diff
@@ -188,24 +200,62 @@ class Playground extends Component<PlaygroundProps, PlaygroundState> {
     )
   }
 
-  render() {
-    const { styles } = this.props
-    const { code, fullscreen, rtl } = this.state
-    const PreviewComponent = ({ themeKey, themes }: AppContextType) => (
+  renderPreview = (code: any, themeKey: any, themes: any) => {
```
Comment (matyasf) [inline packages/__docs__/src/Playground/index.tsx:203]: Can you please type these?

File: packages/__docs__/src/Playground/index.tsx [hunk 6]
```diff
@@ -188,24 +200,62 @@ class Playground extends Component<PlaygroundProps, PlaygroundState> {
...
           fullscreen={fullscreen}
           rtl={rtl}
           themeKey={themeKey}
           themes={themes}
         />
       </PreviewErrorBoundary>
     )
+  }
+
+  renderTabPanel = (themeKey: any, themes: any) => {
```
Comment (matyasf) [inline packages/__docs__/src/Playground/index.tsx:225]: Can you please type these?

**Review by matyasf — CHANGES_REQUESTED** [review]
There is an empty gap between the sample rendering and the edit, codesandbox etc buttons that was not there in the original, please remove it
image
Also there are 2 cases with `any` typing

**Review by balzss — APPROVED** [review]
do we need to display the class/function tab all the time? might be better to only show when users open the source code of the component. otherwise it looks good

**Review by matyasf — APPROVED** [review]
looks good!

**Pull request merged**